server/main.lua

```diff
@@ -23,7 +20,10 @@ lib.callback.register('qbx_scoreboard:server:getScoreboardData', function()
     return totalPlayers, policeCount, players
 end)
 
-RegisterNetEvent('qb-scoreboard:server:SetActivityBusy', function(activity, bool)
-    sharedConfig.illegalActions[activity].busy = bool
-    TriggerClientEvent('qb-scoreboard:client:SetActivityBusy', -1, activity, bool)
-end)
+local function setActivityBusy(name, bool)
+    local illegalActions = GlobalState.illegalActions
+    illegalActions[name].busy = bool
+    GlobalState.illegalActions = illegalActions
```

Comment:
Is illegal actions going to be used by multiple resources? Curious your thoughts behind making it global state.

Reply:
I hadn't thought about sharing the state between resources. That simplifies synchronization between clients and the server. However, as you mention, this state can be used to report new activities to the scoreboard.

Reply:
I mean you could just use exports from the resources returning the amount of needed cops for usage of that resource and then just use `exports.qbx_core:GetDutyCountType('leo')`.

Reply:
I don't know that I'm convinced that using GlobalState for this table is an improvement. This is a UI that currently callbacks to the server to populate the data, a similar concept to loading a web page. Changing to a statebag would always be syncing this data to the clients even though clients wouldn't be needing it most of the time.

Reply:
Does it stress the server or the client? The update is more likely to be only once. Alternatively, the event handler for the state change can be added.

Reply:
In terms of activities reporting what I mean is reporting to qbx_scoreboard name, label and minimum services requirements.

Reply:
The thing that will be changing will be the number of police on duty, so this will be synced fairly often as that number changes, no?

Reply:
Number of police officers on duty is not kept in GlobalStatebag.
The busy parameter can only change when the robbery calls qb-scoreboard:server:SetActivityBusy, the rest of the parameters are unchanged.

Reply:
```mermaid
flowchart LR
    sharedConfig.illegalActions -.init.-> GS[(GlobalState)]
    robbery -.starts.-> SetActivityBusy -.set busy true.-> GS[(GlobalState)]
    robbery -.ends.-> SetActivityBusy -.set busy false.-> GS[(GlobalState)]
```

Reply:
Robbery busy depending on the number of police officers on duty is checked in the UI
```js
if (category.busy)
      $(status).html('<i class="fas fa-clock"></i>');
    else if (data.currentCops >= category.minimumPolice)
      $(status).html('<i class="fas fa-check"></i>');
    else
      $(status).html('<i class="fas fa-times"></i>');
  });
  ```

Reply:
I was thinking of expanding this mechanism to include reporting new actions.
```mermaid
flowchart LR
    sharedConfig.illegalActions -.init.-> GS[(GlobalState)]
    robbery -.init.->registerIllegalAction-.set label, name and requirements.->GS[(GlobalState)]
    robbery -.starts.-> SetActivityBusy -.set busy true.-> GS[(GlobalState)]
    robbery -.ends.-> SetActivityBusy -.set busy false.-> GS[(GlobalState)]
```

Reply:
There's a similar pattern in qbx_management, where we initialize from config, but support adding via export during runtime.

Reply:
Should I convert this to a server-side variable?